osbrain/agent.py

```diff
@@ -1668,10 +1668,13 @@ class AgentProcess(multiprocessing.Process):
     Agent class. Instances of an Agent are system processes which
     can be run independently.
     """
-    def __init__(self, name, nsaddr=None, addr=None, serializer=None,
+    def __init__(self, name=None, nsaddr=None, addr=None, serializer=None,
                  transport=None, base=Agent, attributes=None):
         super().__init__()
-        self.name = name
+        if name:
+            self.name = name
+        else:
+            self.name = unique_identifier().decode()
```

Comment:
Maybe better:

```python
if name is None:
    name = unique_identifier().decode()
```

Anyway this should not be here, as you are not reusing the Agent's UUID. Was not that the idea?

Reply:
Maybe I got this wrong, but this is  `AgentProcess`, which doesn't have a UUID attribute. It's the `Agent` class that has it, so we should probably generate a new UUID and use that instead.

Reply:
Well, you suggested reusing it! :joy:

Anyway I like that idea, and it definitely can be done:

- The `Agent()` class should allow for an empty `name` attribute. If `name is None` then we assign to it the string representation of the Agent's UUID.
- In `AgentProcess`'s `run()` method we can get the `Agent` name attribute just before registering it in the daemon. Then, for registering in the name server we use that name instead of `self.name`.
- Now we need to tell the parent the name that was used. We can pass that information in `self.queue.put('STARTED:name')` (for example).
- Then the `AgentProcess`'s `start()` method should return the name that was used.
- `run_agent()` then knows the name as well and can create the proxy using the returned name instead of the passed `name` parameter.

Reply:
You're right. This seems to work so far.